jax/experimental/jet.py

```diff
@@ -12,6 +12,41 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+r"""Jet is an experimental module for higher-order automatic differentiation
+  that does not rely on repeated first-order automatic differentiation.
+
+  How? Through the propagation of truncated Taylor polynomials.
+  Let :math:`x: \mathbb{R} \rightarrow \mathbb{R}^n` be the Taylor polynomial
+
+  .. math:: x(t) := x_0 + x_1 t + \frac{x_2}{2!} t^2 + ... + \frac{1}{d!} x_d t^d.
+
+  For some function :math:`f: \mathbb{R}^n \rightarrow \mathbb{R}^m`,
+  you can use :func:`jet` to compute the
+  Taylor polynomial
+
+  .. math:: y(t) := y_0 + y_1 t + \frac{y_2}{2!} t^2 + ... + \frac{1}{d!} y_d t^d.
+
+  corresponding to :math:`f(x(\cdot)):\mathbb{R} \rightarrow \mathbb{R}^m`,
+  and relates to :math:`x(t)` and :math:`f(z)` through
+
+  .. math::
+    y_k = \frac{d^k}{dt^k}f(x(t_0)), \quad k=0,...,d.
+
+  More specifically, :func:`jet` computes
+
+  .. math::
+    y_0, (y_1, . . . , y_d) = \texttt{jet} (f, x_0, (x_1, . . . , x_d))
+
+  and can thus be used for high-order
+  automatic differentiation of :math:`f`.
+  Details are explained in
+  `this paper <https://openreview.net/forum?id=SkxEF3FNPH>`__.
+
+  Note:
+    Help improve :func:`jet` by contributing
+    `outstanding primitive rules <https://github.com/google/jax/issues/2431>`__.
```

Comment:
Other experimental modules had similar comments: `ann.py` lists todos, or `host_callback` states "please give feedback, and expect changes.". If I should remove this comment, though, please let me know.

Reply:
This seems like a good idea!